packages/world-module-metadata/test/MetadataModule.t.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { Test } from "forge-std/Test.sol";
+import { GasReporter } from "@latticexyz/gas-report/src/GasReporter.sol";
+
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { createWorld } from "@latticexyz/world/test/createWorld.sol";
+import { ResourceId, WorldResourceIdLib, WorldResourceIdInstance } from "@latticexyz/world/src/WorldResourceId.sol";
+import { IWorldErrors } from "@latticexyz/world/src/IWorldErrors.sol";
+import { NamespaceOwner } from "@latticexyz/world/src/codegen/tables/NamespaceOwner.sol";
+
+import { MetadataModule } from "../src/MetadataModule.sol";
+import { IWorld } from "../src/codegen/world/IWorld.sol";
+import { ResourceTag } from "../src/codegen/tables/ResourceTag.sol";
+
+contract MetadataModuleTest is Test, GasReporter {
+  using WorldResourceIdInstance for ResourceId;
+
+  IWorld world;
+  MetadataModule metadataModule = new MetadataModule();
+
+  function setUp() public {
+    world = IWorld(address(createWorld()));
+    StoreSwitch.setStoreAddress(address(world));
+  }
+
+  function testInstall() public {
+    startGasReport("install metadata module");
+    world.installModule(metadataModule, new bytes(0));
+    endGasReport();
+
+    ResourceId namespace = ResourceTag._tableId.getNamespaceId();
+    assertEq(NamespaceOwner.get(namespace), address(this));
+
+    // Installing again will revert because metadata namespace isn't owned by the module, so the module is unable to write to it.
+    vm.expectRevert(
+      abi.encodeWithSelector(IWorldErrors.World_AccessDenied.selector, namespace.toString(), address(metadataModule))
+    );
+    world.installModule(metadataModule, new bytes(0));
+
+    // Transferring the namespace to the module and installing should be a no-op/idempotent and the module will return namespace ownership.
+    // TODO: is this a security concern? could someone frontrun by putting a tx in between the two calls below?
```

Comment:
you can always make it atomic by doing both via a batchcall. But even without i don't think there is a security concern here, the module would just own the namespace, but unless the module has functions to do something malicious with or is upgradable, another tx in between couldn't do anything else with it